File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const freq = new Map()
        for(const num of nums){
            freq.set(num, (freq.get(num)?? 0) +1)
        }
        const pairs = Array.from(freq.entries()) 
        pairs.sort((a, b) => b[1] - a[1]) 
        return pairs.slice(0, k).map(([num, _]) => num);
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-7.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        pairs = {"}":"{", "]": "[", ")":"("}
        for c in s:
            if c not in pairs:
                stack.append(c)
            else:
                if stack and stack[-1] == pairs[c]:
                    stack.pop()
                else:
                    return False 
        return not stack